code/controllers/subsystem/radioactive_nebula.dm

```diff
@@ -0,0 +1,56 @@
+#define TRAIT_RADIOACTIVE_NEBULA_FAKE_IRRADIATED "radioactive_nebula_fake_irradiated"
+
+/// Controls making objects irradiated when Radioactive Nebula is in effect.
+SUBSYSTEM_DEF(radioactive_nebula)
+	name = "Radioactive Nebula"
+	flags = SS_BACKGROUND
+	wait = 30 SECONDS
+
+	VAR_PRIVATE
+		datum/station_trait/nebula/hostile/radiation/radioactive_nebula
+
+/datum/controller/subsystem/radioactive_nebula/Initialize()
+	radioactive_nebula = locate() in SSstation.station_traits
+	if (!radioactive_nebula)
+		can_fire = FALSE
+		return SS_INIT_NO_NEED
+
+	// We don't *really* care that this happens by the time the server is ready to play.
+	ASYNC
+		irradiate_everything()
+
+	return SS_INIT_SUCCESS
+
+/// Makes something appear irradiated for the purposes of the Radioactive Nebula
+/datum/controller/subsystem/radioactive_nebula/proc/fake_irradiate(atom/movable/target)
+	if (HAS_TRAIT(target, TRAIT_RADIOACTIVE_NEBULA_FAKE_IRRADIATED))
+		return
+
+	ADD_TRAIT(target, TRAIT_RADIOACTIVE_NEBULA_FAKE_IRRADIATED, REF(src))
```

Comment:
There's no logic for removing this trait? Seems like I could enter and exit space to become immune to the nebula

Reply:
You could probably get away with just removing the trait on on_exited() (where the glow is also removed

Reply:
Done